ibm/service/kubernetes/resource_ibm_container_addons.go

```diff
@@ -551,3 +735,30 @@ func resourceIBMContainerAddonsHash(v interface{}) int {
 
 	return conns.String(buf.String())
 }
+
+// Check if correct parameters are provided by the user
+func validateAddonOptions(jsonInput map[string]interface{}, yamlDefault map[string]interface{}) (bool, error) {
+
+	for i, _ := range jsonInput {
+		if _, ok := yamlDefault["data"].(map[string]interface{})[i]; ok {
+			continue
+		} else {
+			return false, fmt.Errorf("[ERROR] Parameter Mismatch")
+		}
+
+	}
+
+	return true, nil
+}
+
+// Set the Value of the Parameters provided by the user in terraform in the config map through index matching
+
+func updateAddOnOptions(jsonInput map[string]interface{}, yamlDefault map[string]interface{}) map[string]interface{} {
```

Comment:
pass the inputs as pointers & we can remove the return statement from this function.